### PR TITLE
Changing return code from storegateway for CMK errors

### PR DESCRIPTION
**Pull request by alanprot**
**What this PR does**:
This is a follow up of https://github.com/cortexproject/cortex/pull/5420 and basically change the return code  from `ResourceExhausted` to `PermissionDenied`.

Also adding the changelog that was forgotten on the last PR.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Tests updated
- [NA] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/storegateway/bucket_stores.go [hunk 1]
```diff
@@ -299,25 +299,22 @@ func (u *BucketStores) Series(req *storepb.SeriesRequest, srv storepb.Store_Seri
 	}
 
 	store := u.getStore(userID)
+	userBkt := bucket.NewUserBucketClient(userID, u.bucket, u.limits)
```
Comment (yeya24) [inline pkg/storegateway/bucket_stores.go:302]: We are initializing a new bkt client every `Series` request? Can we just use `u.bucket` here?

File: pkg/storegateway/bucket_stores_test.go [hunk 2]
```diff
@@ -513,8 +583,11 @@ func queryLabelsNames(stores *BucketStores, userID, metricName string) (*storepb
 	return stores.LabelNames(ctx, req)
 }
 
-func queryLabelsValues(stores *BucketStores, userID, metricName string) (*storepb.LabelValuesResponse, error) {
+func queryLabelsValues(stores *BucketStores, userID, metricName string, start, end int64) (*storepb.LabelValuesResponse, error) {
 	req := &storepb.LabelValuesRequest{
+		Start: start,
+		End:   end,
+		Label: "__name__",
```
Comment (yeya24) [inline pkg/storegateway/bucket_stores_test.go:590]: Let's also make this part of the func signature?

Comment (alanprot) [reply pkg/storegateway/bucket_stores_test.go:590]: Yeah! lets do that!

File: pkg/storegateway/bucket_stores.go (hunk 1 above)
Comment (alanprot) [reply pkg/storegateway/bucket_stores.go:302]: I had to do this to unwrap the error, but creating this here is just creating an object (it does not do anything).

**Review by alvinlin123 — APPROVED** [review]

**Review by yeya24 — APPROVED** [review]

**Pull request merged**